### PR TITLE
use rulesMeta property to get docs URL

### LINKED ISSUE #42: Use the `rulesMeta ` property to get docs URL
[Issuehunt badges](https://issuehunt.io/r/sindresorhus/eslint-formatter-pretty/issues/42)

With a fallback to the existing solution if it doesn't exist.

https://github.com/eslint/eslint/commit/bc3e427ee8875c53eac6b6762884b50074f1adfc

---

<details>
<summary>
<b>IssueHunt Summary</b>
</summary>

### Backers (Total: $60.00)

- [issuehunt issuehunt](https://issuehunt.io/u/issuehunt) ($60.00)

### Submitted pull Requests
- [#44 used rulesMeta property to get docs URL](https://issuehunt.io/r/sindresorhus/eslint-formatter-pretty/pull/44)
- [#48 Use the rulesMeta property to get docs URL](https://issuehunt.io/r/sindresorhus/eslint-formatter-pretty/pull/48)
---

### Tips

- Checkout the [Issuehunt explorer](https://issuehunt.io/r/sindresorhus/eslint-formatter-pretty/) to discover more funded issues.
- Need some help from other developers? [Add your repositories](https://issuehunt.io/r/new) on IssueHunt to raise funds.
---
IssueHunt has been backed by the following sponsors. [Become a sponsor](https://issuehunt.io/membership/members)
</details>

### PR DESCRIPTION
**Pull request by gurrrung**
Fixes #42 
with fallback to the existing solution
if it doesn't exist.

### CODE REVIEW

File: index.js [hunk 1]
```diff
@@ -121,8 +121,13 @@ module.exports = results => {
 
 		if (x.type === 'message') {
 			let ruleUrl;
+
 			try {
-				ruleUrl = getRuleDocs(x.ruleId).url;
+				if (typeof data === 'undefined') {
```
Comment (sindresorhus) [inline index.js]: This check is not correct. `data` can exist even without the `rulesMeta.url` property being available. You need to check for the native `.url` property first and then fall back to `getRuleDocs`. This also needs docs and tests.

---

This issue has a bounty, so it's expected that you put a great effort into the PR.

**Comment by sindresorhus** [comment]
The PR title should follow the same format as the other commits: https://github.com/sindresorhus/eslint-formatter-pretty/commits/master

**Comment by sindresorhus** [comment]
Ping

**Comment by gurrrung** [comment]
How about something like?
				`ruleUrl = data.rulesMeta[x.ruleId].url;
				if (typeof ruleUrl === 'undefined') {
					ruleUrl = getRuleDocs(x.ruleId).url;
				}`

**Comment by gurrrung** [comment]
Screenshot from 2019-07-09 07-49-05

Don't know why I'm getting these eslint errors on running `npm run test` even though my code doesn't have any of these errors.

File: index.js (hunk 1 above)
Comment (gurrrung) [reply index.js]: Actually we are falling back to same solution, thus same tests will do, isn't it?

**Comment by sindresorhus** [comment]
Closing in favor of https://github.com/sindresorhus/eslint-formatter-pretty/pull/48

**Pull request closed without merging**